x/stakeibc/keeper/msg_server_submit_tx.go

```diff
@@ -358,9 +359,28 @@ func (k Keeper) GetLightClientTimeSafely(ctx sdk.Context, connectionID string) (
 }
 
 // Submits an ICQ to get a validator's exchange rate
-func (k Keeper) QueryValidatorExchangeRate(ctx sdk.Context, chainId, validatorAddress string) error {
+func (k Keeper) QueryValidatorExchangeRate(
+	ctx sdk.Context,
+	chainId string,
+	validatorAddress string,
+	callbackDataBz []byte,
+	aggressiveTimeout bool,
```

Comment:
The aggressiveTimeout param feels like code smell of leaky abstraction.  How this method is arrived at (LSM liquid stake or a periodic accounting check) affects its inside implementation behavior through this variable... so someone reading this function has to be aware of the higher level complexities of how it can be called instead of having all context by just looking at the inside of the function itself.  Why not make two versions of this function called LiquidStakingQueryValidatorExchangeRate and PeriodicQueryValidatorExchange rate and avoid the aggressiveTimeout var and many case checkings inside and how we call to it?

Reply:
+1, aggressiveTimeout could be more clear

Reply:
Objective is to avoid having to duplicate code across two functions which then puts a burden on the programmer to keep them in sync.

Maybe a middle ground is something like

```go
func SubmitValidatorExchangeRateICQ(callbackArgs, timeoutPolicy, timeoutDuration) {
   // get host zone
   // get validator
  // build query format
  // submit ICQ
}

func LSMQueryValidatorExchangeRate() {
   SubmitValidatorExchangeRateICQ(...)
}
func PeriodicQueryValidatorExchangeRate() {
   SubmitValidatorExchangeRateICQ(...)
}
```

Reply:
Yes exactly, no need to repeat the core logic but this also splits the behaviors into separate functions with better naming so it is clearer IMO how the function was arrived at compared to before where the condition checking internal to the function was doing this routing

Reply:
🤝

Reply:
I like this solution